pkg/scheduler/server/api.go

```diff
@@ -61,111 +66,93 @@ func (s *Server) ScheduleJob(ctx context.Context, req *schedulerv1pb.ScheduleJob
+			return etcdcron.Failure, nil
+		}
 
-	resp := &schedulerv1pb.ListJobsResponse{Jobs: jobs}
+		return etcdcron.OK, err
+	}
 
-	return resp, nil
+	// Trigger not supported
+	log.Warn("Cannot trigger job: not supported.")
+	return etcdcron.Failure, nil
 }
 
-// GetJob is a placeholder method that needs to be implemented
-func (s *Server) GetJob(ctx context.Context, req *schedulerv1pb.JobRequest) (*schedulerv1pb.GetJobResponse, error) {
+func (s *Server) DeleteJob(ctx context.Context, req *schedulerv1pb.JobRequest) (*schedulerv1pb.DeleteJobResponse, error) {
 	select {
 	case <-ctx.Done():
 		return nil, ctx.Err()
 	case <-s.readyCh:
 	}
 
-	job := s.cron.GetJob(req.GetJobName())
-	if job != nil {
-		resp := &schedulerv1pb.GetJobResponse{
-			Job: &runtimev1pb.Job{
-				Name:     job.Name,
-				Schedule: job.Rhythm,
-				Repeats:  job.Repeats,
-				DueTime:  job.DueTime,
-				Ttl:      job.TTL,
-				Data:     job.Data,
-			},
-		}
-		return resp, nil
+	jobName := composeJobName(req.GetNamespace(), req.GetJobName())
+	err := s.cron.DeleteJob(ctx, jobName)
+	if err != nil {
+		log.Errorf("error deleting job %s: %s", jobName, err)
+		return nil, err
 	}
 
-	return nil, fmt.Errorf("job not found")
+	return &schedulerv1pb.DeleteJobResponse{}, nil
 }
 
-// DeleteJob is a placeholder method that needs to be implemented
-func (s *Server) DeleteJob(ctx context.Context, req *schedulerv1pb.JobRequest) (*schedulerv1pb.DeleteJobResponse, error) {
-	select {
-	case <-ctx.Done():
-		return nil, ctx.Err()
-	case <-s.readyCh:
+// parseStartTime is a wrapper around timeutils.ParseTime that truncates the time to seconds.
+func parseStartTime(dueTime string) (time.Time, error) {
+	if dueTime == "" {
+		return time.Time{}, nil
 	}
 
-	err := s.cron.DeleteJob(req.GetJobName())
+	now := time.Now()
+	t, err := timeutils.ParseTime(dueTime, &now)
 	if err != nil {
-		log.Errorf("error deleting job %s: %s", req.GetJobName(), err)
-		return nil, err
+		return t, err
 	}
-
-	return &schedulerv1pb.DeleteJobResponse{}, nil
+	t = t.Truncate(time.Second)
+	return t, nil
 }
 
-func (s *Server) TriggerJob(ctx context.Context, req *schedulerv1pb.TriggerJobRequest) (*schedulerv1pb.TriggerJobResponse, error) {
-	log.Info("Triggering job")
-	metadata := req.GetMetadata()
-	actorType := metadata["actorType"]
-	actorID := metadata["actorId"]
-	reminderName := metadata["reminder"]
-	if actorType != "" && actorID != "" && reminderName != "" {
-		if s.actorRuntime == nil {
-			return nil, fmt.Errorf("actor runtime is not configured")
-		}
+func parseTTL(ttl string) (time.Duration, error) {
+	if ttl == "" {
+		return time.Duration(0), nil
+	}
 
-		invokeMethod := "remind/" + reminderName
-		contentType := metadata["content-type"]
-		invokeReq := internalv1pb.NewInternalInvokeRequest(invokeMethod).
-			WithActor(actorType, actorID).
-			WithData(req.GetData().GetValue()).
-			WithContentType(contentType)
+	years, months, days, period, _, err := timeutils.ParseDuration(ttl)
```

Comment:
Please can we make `timeutils.ParseDuration(ttl)` return a `struct`. Returning a list of similar typed variables is a source of bugs.

Reply:
Yes, we will. I am using the Dapr code as-is for now.